presidio-analyzer/analyzer/analyzer_engine.py

```diff
@@ -0,0 +1,92 @@
+import logging
+import analyze_pb2
+import analyze_pb2_grpc
+import common_pb2
+import os
+
+from analyzer import RecognizerRegistry
+from analyzer import ContextEnhancer
+
+loglevel = os.environ.get("LOG_LEVEL", "INFO")
+logging.basicConfig(
+    format='%(asctime)s:%(levelname)s:%(message)s', level=loglevel)
+
+DEFAULT_LANGUAGE = "en"
+
+
+class AnalyzerEngine(analyze_pb2_grpc.AnalyzeServiceServicer):
+    def __init__(self, registry=RecognizerRegistry()):
+        # load all recognizers
+        self.registry = registry
+        self.context_enhancer = ContextEnhancer()
+        registry.load_local_recognizer("predefined-recognizers")
+
+    @staticmethod
+    def __remove_duplicates(results):
+
+        filtered_results = []
+
+        for result in results:
+            valid_result = True
+            if result not in filtered_results:
+                for filtered in filtered_results:
+                    # If result is equal to or substring of a checksum result
+                    if result.start >= filtered.start and result.end <= filtered.end:
+                        valid_result = False
```

Comment:
Will this work if my first result has a small span and the second has a large span? we will return them both. 
Maybe sorting by span size before this process could help (or some other logic that deals with this)
Right now it is sorted by (start, score) which still can cause this if both spans start at the same index (If I understand correctly)

Reply:
We opened a bug to track this issue.